lib/DDG/Spice/Libraries.pm

```diff
@@ -0,0 +1,22 @@
+package DDG::Spice::Libraries;
+# ABSTRACT: Search for code on libraries.io
+
+use strict;
+use DDG::Spice;
+
+my @triggers = share('triggers.txt')->slurp;
+triggers startend => @triggers;
+
+my $words = join "|", @triggers;
+$words =~ s/\n//g;
+
+spice from => '(.+)/(.+)';
+spice to => 'https://libraries.io/api/search?platforms=$1&q=$2&api_key={{ENV{DDG_SPICE_LIBRARIES_IO_APIKEY}}}';
+spice wrap_jsonp_callback => 1;
+
+handle remainder_lc => sub {
+    return unless $_;
+    return $req->matched_trigger, $_;
```

Comment:
@knowledgewarrior sorry for the confusion, I meant this was more of a starting point. We need to map the value of `matched_query` to the appropriate platform.

There are a few ways to do that, I think the easiest is to build a hash mapping triggers that don't match a platform name, to the appropriate platform, and do a lookup in that hash using the `matched_trigger`. If it fails, you can fallback to using the `matched_trigger` value, as it should be a platform name:

```perl
my %platforms = (
  "go get" => "go",
  "elm package" => "elm",
  ...
);

...

handle remainder_lc => sub {
    return unless $_;
    my $trigger = $req->matched_trigger;
    my $platform = $platforms{$trigger} // $trigger;
    return $platform, $_;
}
```

Does that make sense?